### ISSUE TITLE
Add guidance for "regular expression" (regex)

### ISSUE THREAD

**Issue by kcarmichael08** [redhat-documentation/supplementary-style-guide#226]
There does not appear to be official guidance regarding use of the term "regular expression" (also known as "regex"), which is commonly used in searches, cron jobs, etc. Michael Peter found a mention in Google's Style Guide: https://developers.google.com/style/word-list#letter-r

Slack thread: https://coreos.slack.com/archives/C0227BTM07K/p1663694176718289

More info on regular expression: https://en.wikipedia.org/wiki/Regular_expression

**Comment by bburt-rh** [comment]
I agree that we need guidance for this. One possibility would be to add two entries to the glossary:

- `regex: Do not use. Instead use regular expression`
- `regular expression:` [definition of the phrase]

Or might it be better to say the `regex` is OK to use on second use, but first use should be `regular expression (regex)`

@redhat-documentation/ccs-style-council What do others think?

**Comment by Amrita42** [comment]
I like this option >> regular expression: [definition of the phrase]. Use instead of regex.  I see more usage of `regular expression` in https://docs.oracle.com/cd/B19306_01/B14251_01/adfns_regexp.htm#CHDDFAFA.

**Comment by bergerhoffer** [comment]
I'm leaning toward saying not to use "regex". It might be fine, as long as you write out out it first. But it seems like more of a casual way to refer to it, and probably doesn't belong that way in product docs. So unless we have a good reason why it is appropriate in some cases, then I'd say we should follow suit with Google and say to not use it.

**Comment by kcarmichael08** [comment]
Yeah, I spelled it out and added the abbreviation in the edge case I had, but that's because `regex` was used in the bug description that I was writing about. Normally I would say, don't use.

**Comment by Amrita42** [comment]
Found this ref >> https://developers.redhat.com/articles/2022/09/14/beginners-guide-regular-expressions-grep?sc_cid=7013a000003899eAAA

**Comment by dfitzmau** [comment]
If you type `man 7 regex` into your RHEL system's terminal, you'll access the Linux Programmer's Manual that describes regular expressions and the `regex` package. I agree with using "regular expression" for general discussion points and only use `regex` when referring to the package itself and so on.

**Comment by bergerhoffer** [comment]
@kcarmichael08 It sounds like we're pretty much in agreement to not use "regex" and instead use "regular expression".

Would you be willing to open a PR to add this, or would you like someone from the SSG to work on it?

I'd recommend two entries like Brian recommends above: a "regex" entry with "Use it: no" and a "regular expression" with "Use it: yes".

**Comment by kcarmichael08** [comment]
> @kcarmichael08 It sounds like we're pretty much in agreement to not use "regex" and instead use "regular expression".
> 
> Would you be willing to open a PR to add this, or would you like someone from the SSG to work on it?
> 
> I'd recommend two entries like Brian recommends above: a "regex" entry with "Use it: no" and a "regular expression" with "Use it: yes".

I'll do it - good practice :)

**Comment by bburt-rh** [comment]
Thanks @kcarmichael08! When you create the PR, can you please link it to this issue? That way, when the PR is merged, GH will automatically close this issue. Thanks again!

**Comment by sbmetz** [comment]
@kcarmichael08 -- any progress on this PR? LMK if I can help!

**Comment by kcarmichael08** [comment]
> @kcarmichael08 -- any progress on this PR? LMK if I can help!

Sorry, I have been swamped. :(  If I should let someone else do the PR, that is fine, or I will try to get to it by the end of the week.

**Comment by bredamc** [comment]
PR 255 has been merged, so I'm closing this issue. Thank you @kcarmichael08 !
